src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -98,15 +98,19 @@ protected override ModuleData CreateValueFromKey(string key)
         private SimpleNameHashtable _simpleNameHashtable = new SimpleNameHashtable();
 
         private SharedGenericsMode _genericsMode;
+
+        private readonly DelegateInfoCreator _delegateInfoCreator;
         
-        public CompilerTypeSystemContext(TargetDetails details, SharedGenericsMode genericsMode)
+        public CompilerTypeSystemContext(TargetDetails details, SharedGenericsMode genericsMode, DelegateInfoCreator delegateInfoCreator)
```

Comment:
It does not feel right to me to have `DelegateInfoCreator` injected as a dependency into CompilerTypeSystemContext.

Would it be better have a "Features" enums passed to CompilerTypeSystemContext that we use for this, and maybe other similar things in future? (Feel free to say "no".)

Reply:
I started with that, but I wasn't quite sure how to call this feature and what sort of granularity we would want to expose it at (do we expose all delegate non-mandatory thunk kinds?).

If you have a suggestion in that area, I would be more than willing to take it. I don't particularly like the `DelegateInfoCreator` either.

Reply:
What about just using the existing `DelegateFeature` enum for this, like:

`public CompilerTypeSystemContext(TargetDetails details, SharedGenericsMode genericsMode, DelegateFeature delegateFeatures)` ?

Reply:
Yup, that works.